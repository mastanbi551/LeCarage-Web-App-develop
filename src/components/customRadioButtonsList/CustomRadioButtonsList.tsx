import React, { useState, useEffect } from 'react';

interface RadiobuttonListInfo {
  value: string;
  label: string;
}
interface Props {
  name: string | undefined;
  radioButtonList: RadiobuttonListInfo[];
  handleRadioButtonLClick: (e: React.ChangeEvent<HTMLInputElement>,
    featureValue: string,
    feature:  {
      value: string,
      label: string
    },
    access: {
      value: string,
      label: string
    }) => void;
  isDisable?: boolean;
  error?: string;
  defaultValue?: string;
  showSelectText?: boolean;
  onBlur?: (e: React.FocusEvent<HTMLSelectElement> | undefined) => void;
  touched?: boolean;
  id: string;
  defaultChecked?: string | number;
  values: {featureId: string, accessId: string}[];
  feature: {value: string, label: string};
}

export const CustomRadioButtonList: React.FC<Props> = ({
  name,
  radioButtonList,
  handleRadioButtonLClick,
  isDisable,
  id,
  defaultChecked,
  values,
  feature,
}) => {
  const [checkedorNot, setCheckedOrNot] = useState(defaultChecked);

  useEffect(() => {
    if(values.length === 0){
      setCheckedOrNot(-1);
    }
  }, [values]);

  return (
    <div className='radiobutton row'>
      {/* {labelName && <label className='radiobutton-label'>{labelName}</label>} */}
      {radioButtonList &&
        radioButtonList.map((access, index) => {
          return values.length === 0 ? (
            <div className='col-md-3 feature-radio' key={index}>
              <input
                type='radio'
                id={`feature_${id}_access_${access.value}`}
                name={`feature_${id}_access`}
                value={access.value}
                onChange={(e) => {
                  setCheckedOrNot(access.value);
                  handleRadioButtonLClick(e, id, feature, access);
                }}
                disabled={isDisable}
                className={name}
                checked={
                  values.length === 0 && checkedorNot === access.value
                    ? true
                    : false
                }
              />
              <label htmlFor={`feature_${id}_access_${access.value}`}>
                {access.label}
              </label>
            </div>
          ) : (
            <div className='col-md-3 feature-radio' key={index}>
              <input
                type='radio'
                id={`feature_${id}_access_${access.value}`}
                name={`feature_${id}_access`}
                value={access.value}
                onChange={(e) => {
                  setCheckedOrNot(access.value);
                  // setCheckedOrNot(-1);
                  handleRadioButtonLClick(e, id, feature, access);
                }}
                disabled={isDisable}
                className={name}
                checked={
                  values.length > 0 && checkedorNot === access.value
                    ? true
                    : false
                }
              />
              <label htmlFor={`feature_${id}_access_${access.value}`}>
                {access.label}
              </label>
            </div>
          );
        })}
    </div>
  );
};
