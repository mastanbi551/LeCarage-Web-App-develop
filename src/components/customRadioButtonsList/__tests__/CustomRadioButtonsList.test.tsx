import React from 'react';
import { render } from '@testing-library/react';
import { CustomRadioButtonList } from '../CustomRadioButtonsList';

describe('CustomRadioButtonList', () => {
  it('renders correctly', () => {
    const props = {
      name: 'test',
      radioButtonList: [
        { value: '1', label: 'Option 1' },
        { value: '2', label: 'Option 2' },
        { value: '3', label: 'Option 3' },
      ],
      handleRadioButtonLClick: jest.fn(),
      values: [],
      feature: { value: 'test', label: 'Test' },
    };
    const { container } = render(<CustomRadioButtonList {...props} />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
