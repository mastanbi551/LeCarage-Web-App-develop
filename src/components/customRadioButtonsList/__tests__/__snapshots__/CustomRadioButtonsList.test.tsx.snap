// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomRadioButtonList renders correctly 1`] = `
<div
  class="radiobutton row"
>
  <div
    class="col-md-3 feature-radio"
  >
    <input
      class="test"
      id="feature_undefined_access_1"
      name="feature_undefined_access"
      type="radio"
      value="1"
    />
    <label
      for="feature_undefined_access_1"
    >
      Option 1
    </label>
  </div>
  <div
    class="col-md-3 feature-radio"
  >
    <input
      class="test"
      id="feature_undefined_access_2"
      name="feature_undefined_access"
      type="radio"
      value="2"
    />
    <label
      for="feature_undefined_access_2"
    >
      Option 2
    </label>
  </div>
  <div
    class="col-md-3 feature-radio"
  >
    <input
      class="test"
      id="feature_undefined_access_3"
      name="feature_undefined_access"
      type="radio"
      value="3"
    />
    <label
      for="feature_undefined_access_3"
    >
      Option 3
    </label>
  </div>
</div>
`;
